tests/test_authentication.py

```diff
@@ -140,13 +142,19 @@ def test_unknown_creds(self):
                               '^access denied: CredentialsLookupError: ')
 
     def test_alternative_credential_lookup(self):
-        sender = self._sender(credentials=ALTERNATIVE_CREDS)
-        req = self._request(sender)
         lookup_path = "%s.alternative_lookup" % __name__
         with self.settings(HAWK_CREDENTIALS_LOOKUP=lookup_path):
+            # Reload with settings change, re-instantiate
+            reload_module(hawkrest)
```

Comment:
Right, almost forgot: this was also a consequence of me moving the settings check to module load time instead of runtime. Except for testing, settings are considered immutable, so I saw no reason for keeping a runtime check (especially every call, not just instantiation). Thus `reload_module()`.

New version of the test subclasses `HawkAuthentication` instead, which is cleaner in every way, IMO.